pkg/engine/task/task_apply.go

```diff
@@ -102,15 +104,16 @@ func patch(newObj runtime.Object, existingObj runtime.Object, c client.Client) e
 		// strategic merge patch is not supported for these types, falling back to merge patch
 		err := c.Patch(context.TODO(), newObj, client.ConstantPatch(types.MergePatchType, newObjJSON))
 		if err != nil {
-			return fmt.Errorf("failed to apply merge patch to object %s/%s: %w", key.Name, key.Name, err)
-		}
-	} else {
-		err := c.Patch(context.TODO(), existingObj, client.ConstantPatch(types.StrategicMergePatchType, newObjJSON))
-		if err != nil {
-			return fmt.Errorf("failed to apply StrategicMergePatch to object %s/%s: %w", key.Namespace, key.Name, err)
+			return nil, fmt.Errorf("failed to apply merge patch to object %s/%s: %w", key. Namespace, key.Name, err)
 		}
+		return newObj, nil
 	}
-	return nil
+
+	err := c.Patch(context.TODO(), existingObj, client.ConstantPatch(types.StrategicMergePatchType, newObjJSON))
```

Comment:
It's not an error, but makes it hard to read: in this Patch-call you use `existingObj` to get the updated object back, and in the Patch-call above it's `newObj`.

Reply:
yeah I know, what do you propose? 🤔 this is just a confusion in the patch API of the client 😢

Reply:
> yeah I know, what do you propose?

using the same var in both places e.g. `newObj`?